Modules/Core/Common/include/itkImageSink.hxx

```diff
@@ -0,0 +1,287 @@
+
+
+  this->GetRegionSplitter()->GetSplit( inputRequestedRegionNumber,
+                                       this->GetNumberOfInputRequestedRegions( ),
+                                       inputImageRegion );
+  m_CurrentInputRegion = inputImageRegion;
+
+  itkDebugMacro( "Generating " << inputRequestedRegionNumber << " chunk as " <<  m_CurrentInputRegion );
+
+
+  for (auto &inputName: this->GetInputNames())
+    {
+    if ( this->ProcessObject::GetInput(inputName) )
+      {
+      // Check whether the input is an image of the appropriate
+      // dimension (use ProcessObject's version of the GetInput()
+      // method sinke it returns the input as a pointer to a
+      // DataObject as opposed to the subclass version which
+      // static_casts the input to an TInputImage).
+      using ImageBaseType = ImageBase< InputImageDimension >;
+      typename ImageBaseType::ConstPointer constInput =
+        dynamic_cast< ImageBaseType const * >( this->GetInput(inputName) );
+
+      // If not an image, skip it, and let a subclass of
+      // ImageToImageFilter handle this input.
+      if ( constInput.IsNull() )
+        {
+        continue;
+        }
+
+      // Input is an image, cast away the constness so we can set
+      // the requested region.
+      InputImagePointer input = const_cast< TInputImage * >( this->GetInput(inputName) );
+
+      // copy the requested region of the first input to the others
+      InputImageRegionType inputRegion;
+      input->SetRequestedRegion( m_CurrentInputRegion );
+      }
+    }
+}
+
+
+template <class TInputImage>
+void
+ImageSink<TInputImage>
+::VerifyInputInformation() ITKv5_CONST
+{
+  using ImageBaseType = const ImageBase< InputImageDimension >;
+
+  ImageBaseType *inputPtr1 = nullptr;
+  InputDataObjectConstIterator it(this);
+
+  for(; !it.IsAtEnd(); ++it )
+    {
+    // Check whether the output is an image of the appropriate
+    // dimension (use ProcessObject's version of the GetInput()
+    // method since it returns the input as a pointer to a
+    // DataObject as opposed to the subclass version which
+    // static_casts the input to an TInputImage).
+    inputPtr1 = dynamic_cast< ImageBaseType * >( it.GetInput() );
+
+    if ( inputPtr1 )
+      {
+      break;
+      }
+    }
+
+    for (; !it.IsAtEnd(); ++it )
+    {
+    auto * inputPtrN = dynamic_cast< ImageBaseType * >( it.GetInput() );
+
+    // Physical space computation only matters if we're using two
+    // images, and not an image and a constant.
+    if( inputPtrN )
+      {
+      // check that the image occupy the same physical space, and that
+      // each index is at the same physical location
+
+      // tolerance for origin and spacing depends on the size of pixel
+      // tolerance for directions a fraction of the unit cube.
+      const SpacePrecisionType coordinateTol
+        = std::abs(this->m_CoordinateTolerance * inputPtr1->GetSpacing()[0]); // use first dimension spacing
```

Comment:
why is there a need for abs here? I expected tolerance and spacing to be positive. The tolerance is double but I expect that when it is set, ITK either forces abs or throws an exception if less than zero.

Reply:
This code is just a dup of what's in ImageToImage. It improves robustness with no downside.